analyze-scope.js

```diff
@@ -286,6 +300,41 @@ class Referencer extends OriginalReferencer {
         this.typeMode = upperTypeMode;
     }
 
+    /**
+     * Visit new expression.
+     * @param {NewExpression} node The NewExpression node to visit.
+     * @returns {void}
+     */
+    NewExpression(node) {
+        this.visitTypeParameters(node);
+        this.visit(node.callee);
+        if (node.arguments) {
+            node.arguments.forEach(this.visit, this);
+        }
+    }
+
+    /**
+     * Override.
+     * Visit call expression.
+     * @param {CallExpression} node The CallExpression node to visit.
+     * @returns {void}
+     */
+    CallExpression(node) {
+        this.visitTypeParameters(node);
+
+        // Check this is direct call to eval
+        if (!this.scopeManager.__ignoreEval() && node.callee.type === "Identifier" && node.callee.name === "eval") {
+
+            // NOTE: This should be `variableScope`. Since direct eval call always creates Lexical environment and
+            // let / const should be enclosed into it. Only VariableDeclaration affects on the caller's environment.
+            this.currentScope().variableScope.__detectEval();
+        }
```

Comment:
i don't like this, but CallExpression is sadly doing visitChildren and iterating over all properties from visitor-keys